lib/metadata/models/object/ObjectMD.js

```diff
@@ -0,0 +1,613 @@
+     */
+    getKey() {
+        return this._data.key;
+    }
+
+    /**
+     * Set location
+     *
+     * @param {string[]} location - location
+     * @return {ObjectMD} itself
+     */
+    setLocation(location) {
+        this._data.location = JSON.parse(JSON.stringify(location));
+        return this;
+    }
+
+    /**
+     * Returns location
+     *
+     * @return {string[]} location
+     */
+    getLocation() {
+        return this._data.location;
+    }
+
+    /**
+     * Set metadata isNull value
+     *
+     * @param {boolean} isNull - Whether new version is null or not
+     * @return {ObjectMD} itself
+     */
+    setIsNull(isNull) {
+        this._data.isNull = isNull;
+        return this;
+    }
+
+    /**
+     * Get metadata isNull value
+     *
+     * @return {boolean} Whether new version is null or not
+     */
+    getIsNull() {
+        return this._data.isNull;
+    }
+
+    /**
+     * Set metadata nullVersionId value
+     *
+     * @param {string} nullVersionId - The version id of the null version
+     * @return {ObjectMD} itself
+     */
+    setNullVersionId(nullVersionId) {
+        this._data.nullVersionId = nullVersionId;
+        return this;
+    }
+
+    /**
+     * Get metadata nullVersionId value
+     *
+     * @return {string} The version id of the null version
+     */
+    getNullVersionId() {
+        return this._data.nullVersionId;
+    }
+
+    /**
+     * Set metadata isDeleteMarker value
+     *
+     * @param {boolean} isDeleteMarker - Whether object is a delete marker
+     * @return {ObjectMD} itself
+     */
+    setIsDeleteMarker(isDeleteMarker) {
+        this._data.isDeleteMarker = isDeleteMarker;
+        return this;
+    }
+
+    /**
+     * Get metadata isDeleteMarker value
+     *
+     * @return {boolean} Whether object is a delete marker
+     */
+    getIsDeleteMarker() {
+        return this._data.isDeleteMarker;
+    }
+
+    /**
+     * Set metadata versionId value
+     *
+     * @param {string} versionId - The object versionId
+     * @return {ObjectMD} itself
+     */
+    setVersionId(versionId) {
+        this._data.versionId = versionId;
+        return this;
+    }
+
+    /**
+     * Get metadata versionId value
+     *
+     * @return {boolean} The object versionId
+     */
+    getVersionId() {
+        return this._data.versionId;
+    }
+
+    /**
+     * Set tags
+     *
+     * @param {object} tags - tags object
+     * @return {ObjectMD} itself
+     */
+    setTags(tags) {
+        this._data.tags = tags;
+        return this;
+    }
+
+    /**
+     * Returns tags
+     *
+     * @return {object} tags object
+     */
+    getTags() {
+        return this._data.tags;
+    }
+
+    /**
+     * Set custom meta headers
+     *
+     * @param {object} data - Meta headers
+     * @return {ObjectMD} itself
+     */
+    setUserMetadata(data) {
+        Object.keys(data).forEach(key => {
+            if (key.startsWith('x-amz-meta')) {
```

Comment:
minor: shouldn't the prefix checked against `x-amz-meta-` instead? (with trailing `-`) not sure it really matters but it looks more correct - could suddenly matter e.g. if amazon decided to use some `x-amz-metacool-header` to annoy devs :rofl: